workbench-core/example/app/src/routes/datasetRoutes.ts

```diff
@@ -57,14 +59,14 @@ export function setUpDSRoutes(
 
   // share dataset
   router.post(
-    '/datasets/:id/share',
+    '/datasets/:datasetId/share',
     wrapAsync(async (req: Request, res: Response) => {
-      if (req.params.id.match(uuidWithLowercasePrefixRegExp(resourceTypeToKey.dataset)) === null) {
-        throw Boom.badRequest('id request parameter is invalid');
+      if (req.params.datasetId.match(uuidWithLowercasePrefixRegExp(dataSetPrefix)) === null) {
+        throw Boom.badRequest('datasetid request parameter is invalid');
```

Comment:
Would it make sense to throw 404's here when a param isnt a valid UUID to prevent attackers from finding out info about how things are stored? Or since if a request gets here, it made it through AuthN/AuthZ so we arent as concerned with attackers?

Reply:
The latter. We actually never make it through authz if the parameter is not in the correct format. If we did make it here, I would actually like to let the authorized user know what they did wrong rather than confuse them with a not found error.